src/coreclr/nativeaot/Runtime/unix/UnwindHelpers.cpp

```diff
@@ -755,6 +755,206 @@ void Registers_REGDISPLAY::setVectorRegister(int num, libunwind::v128 value)
     D[num] = (uint64_t)value.vec[2] << 32 | (uint64_t)value.vec[3];
 }
 
+/// CompactUnwinder_arm64 uses a compact unwind info to virtually "step" (aka
+/// unwind) by modifying a Registers_arm64 register set
+template <typename A>
```

Comment:
llvm-libunwind is only used for NativeAOT, you can make the required changes there. This way we can keep track of how much we have diverged from upstream. Long term plan is to upstream the changes (#72655).

Reply:
I can easily do that. Not sure how much it is frowned upon in regard to local vs upstreamed changes.

Reply:
I didn't meant to upstream the change, just move it under `src/native/external/llvm-libunwind` instead of copy/pasting this whole class.

Reply:
Sure, I will do that. I was just not sure what's the policy about modifying the code in `src/native/external/llvm-libunwind` at all...

Reply:
It goes like this: make a change in `external/{xyz}/` in a separate commit and add `Apply {commit-sha}` line in `/src/native/external/{xyz}-version.txt`. That's all. :)

next to: https://github.com/dotnet/runtime/blob/a62d237af3608d846ffcb0e4d2edc4eca09bbd0a/src/native/external/llvm-libunwind-version.txt#L5

Reply:
So I essentially need to submit one PR for the `external/...` change and then second one for `llvm-libunwind-version.txt` update (since I don't know the commit hash until it's merged on top of `main`)?

Reply:
It can be done in the same PR, just push separate commit(s). Those commits won't be lost. This way we can reconstruct the dependency tree by looking at the version.txt file like so:

```sh
$ cd src/native/external/llvm-libunwind; cd $_
$ git clone https://github.com/llvm/llvm-libunwind --single-branch --depth 1 - b llvmorg-15.0.0 # or whatever
$ rm -rf .git; cd ../../../../
$ curl -sSL https://github.com/dotnet/runtime/commit/e57194552327bccaf2b45a9190f0e6e1289472e4.patch |\
     git apply -
$ curl -sSL https://github.com/dotnet/runtime/commit/db7ed089a9e20a391a53e0b547f95dc8bf916765.patch |\
     git apply -
```

It helps when updating the dependency (and changes are not upstreamed). It also keeps track of things to be upstreamed.

Reply:
Ah, I see. If you fetch it from GitHub it goes directly to the über-repo that has commits from all forks.